### PR TITLE
[git-webkit] Clear merging-blocked label on PR update

### PR DESCRIPTION
**Pull request by JonWBedard**
#### 4496b5ee21d92e9649097e54169bec171cdb0ec4
<pre>
[git-webkit] Clear merging-blocked label on PR update
<a href="https://bugs.webkit.org/show_bug.cgi?id=238907">https://bugs.webkit.org/show_bug.cgi?id=238907</a>
&lt;rdar://problem/91380962 &gt;

Reviewed by Aakash Jain.

* Tools/Scripts/libraries/webkitbugspy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
(Tracker.set): Empty list should trigger request.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py:
(Issue.set_labels): Pass self to tracker.set.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py:
(GitHub.__init__): Set self.labels from passed argument.
(GitHub._issue): Handle empty list.
* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest): Add BLOCKED_LABEL constant.
(PullRequest.main): Remove &apos;merging-blocked&apos; label from existing pull-request
before updating pull-request.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/pull_request_unittest.py:

Canonical link: <a href="https://commits.webkit.org/249376@main">https://commits.webkit.org/249376@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@292538">https://svn.webkit.org/repository/webkit/trunk@292538</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

**Comment by JonWBedard** [comment]
You can see in the history of this PR it functioning on itself.

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/pull_request_unittest.py [hunk 1]
```diff
@@ -394,21 +394,29 @@ def test_github_draft(self):
...
+            'merging-blocked': dict(color='c005E5', description='Applied to prevent a change from being merged'),
+        }) as remote, mocks.local.Git(self.path, remote='https://{}'.format(remote.remote)) as repo, mocks.local.Svn():
             with OutputCapture():
                 repo.staged['added.txt'] = 'added'
                 self.assertEqual(0, program.main(
                     args=('pull-request', '-i', 'pr-branch'),
                     path=self.path,
                 ))
 
+            gh_tracker = github.Tracker('https://{}'.format(remote.remote))
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/pull_request_unittest.py]: gh_tracker => github_tracker

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py [hunk 2]
```diff
@@ -173,6 +174,30 @@ def main(cls, args, repository, **kwargs):
...
+            if existing_pr and not existing_pr.opened and not args.defaults and (args.defaults is False or Terminal.choose(
+                    "'{}' is already associated with '{}', which is closed.\nWould you like to create a new pull-request?".format(
+                        repository.branch, existing_pr,
+                    ), default='No',
+            ) == 'Yes'):
+                existing_pr = None
+
+        # Remove "merging-blocked" label
+        if existing_pr and existing_pr._metadata and existing_pr._metadata.get('issue'):
+            log.info("Checking PR labels for '{}'...".format(cls.BLOCKED_LABEL))
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py]: Do we need to print this at info level?

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py [hunk 3]
```diff
@@ -173,6 +174,30 @@ def main(cls, args, repository, **kwargs):
         if not rmt:
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py]: rmt => remote or _remote or remote_repo

**Review by aj062 — APPROVED** [review]

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py (hunk 2 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py]: In this script, the `info` level is the first `-v`, so unless folks are adding logging flags, they won't see this line.

**Pull request merged**